learn/experimental/vector_search.mdx

```diff
@@ -102,14 +103,86 @@ curl \
 `documentTemplate` is an optional field you can use to customize the data you send to the embedder. It is highly recommended you configure a custom template for your documents.
 
 `model` is mandatory and must indicate a BERT embedding model.
+</Tabs.Content>
+
+<Tabs.Content label="Ollama">
+Use the `embedders` index setting of the [update `/settings` endpoint](/reference/api/settings) to configure one or more embedders for an index:
+
+```sh
+curl \
+  -X PATCH 'http://localhost:7700/indexes/movies/settings' \
+  -H 'Content-Type: application/json' \
+  --data-binary '{
+    "embedders": {
+      "default": {
+        "source": "ollama",
+        "url": "http://localhost:11434/api/embeddings",
+        "apiKey": "OLLAMA_API_KEY",
+        "model": "MODEL_NAME",
+        "documentTemplate": "A document titled '{{doc.title}}' whose description starts with {{doc.overview|truncatewords: 20}}"
+      }
+    }
+  }'
+```
+
+`url` indicates the embedder address. If not specified, Meilisearch will try to fetch the `MEILI_OLLAMA_URL` environment variable.
+
+`apiKey` is an optional field indicating your Ollama API key, if you have configured it.
+
+`model` is a mandatory field indicating a compatible model.
+
+`documentTemplate` is an optional field. Use it to customize the data you send to the embedder. It is highly recommended you configure a custom template for your documents.
+</Tabs.Content>
+
+<Tabs.Content label="REST">
+The generic REST source is compatible with any embedder with a REST API without a dedicated Meilisearch `source`.
+
+Use the `embedders` index setting of the [update `/settings` endpoint](/reference/api/settings) to configure one or more embedders for an index:
+
+```sh
+curl \
+  -X PATCH 'http://localhost:7700/indexes/movies/settings' \
+  -H 'Content-Type: application/json' \
+  --data-binary '{
+    "embedders": {
+      "default": {
+        "source": "rest",
+        "url": "http://localhost:12345/api/v1/embed",
+        "apiKey": "SOURCE_API_KEY",
+        "dimensions": 512,
+        "documentTemplate": "A document titled '{{doc.title}}' whose description starts with {{doc.overview|truncatewords: 20}}",
+        "inputField": ["data", "text"],
+        "inputType": "text",
+        "query": {
+          "model": "MODEL_NAME",
+          "dimensions": 512
+        },
+        "pathToEmbeddings": ["data"],
+        "embeddingObject": ["embedding"]
```

Comment:
Do we plan to define what `data`, `text` or other field means?

Reply:
If you think it's fundamental to explain it, I'm happy to add definitions. Two things, though:

1. I couldn't find them in the reference document, so I'd need someone to explain them to me (or point me to a place where I can find that information)
2. This document is already super bloated. Really, it's a huge bag of "everything related to vector search", so I'm trying not to add more information than strictly necessary. My plan is to soon break this down in more focussed articles

Reply:
I have discussed this with @macraig and we decided to leave this PR without any definitions for the moment. Once both louis and myself are back from our holidays, I'll create a new PR adding more information.